src/bin/pg_autoctl/keeper.c

```diff
@@ -2204,6 +2204,42 @@ keeper_reload_configuration(Keeper *keeper, bool firstLoop, bool doInit)
 }
 
 
+/*
+ * reread_services reads the possibly updated services section from the
+ * configuration file and integrates accepted new values into the current setup.
+ */
+bool
+keeper_reread_services(Keeper *keeper)
+{
+	KeeperConfig *config = &(keeper->config);
+	KeeperConfig newConfig = { 0 };
+
+	if (!file_exists(config->pathnames.config))
+	{
+		log_error("Missing configuration file %s", config->pathnames.config);
+		return false;
+	}
+
+	/*
+	 * Set the same configuration and state file as the current config.
+	 */
+	strlcpy(newConfig.pathnames.config, config->pathnames.config, MAXPGPATH);
+	strlcpy(newConfig.pathnames.state, config->pathnames.state, MAXPGPATH);
+
+	if (!keeper_config_read_file(&newConfig,
+								 true, /* missingPgdataIsOk */
+								 true, /* pgIsNotRunningIsOk */
+								 true /* monitorDisabledIsOk */))
+	{
+		log_error("Failed to read new config");
+		return false;
+	}
+
+	/* update the configuration values (None at present) */
+	return true;
```

Comment:
Correct, at the moment there is no result yet. It is just the scaffolding. You can see how it will behave with the addition of dynamic services here:
https://github.com/citusdata/pg_auto_failover/blob/bb5298b8617305cde5c5551e2327e0650d6f4c00/src/bin/pg_autoctl/keeper.c#L2217

Reply:
I can add the actual code ifdef'ed if you don't mind. That will make it much much easier